tests/apiserver-availability-test.yaml

```diff
@@ -0,0 +1,41 @@
+rule_files:
+- ../prometheus_alerts.yaml
+- ../prometheus_rules.yaml
+
+evaluation_interval: 1m
+
+tests:
+- name: calculate apiserver request total increase 1h rate
+  interval: 1m
+  input_series:
+  # 100 requests in the overall interval, 99 successful and 1 error
+  - series: 'apiserver_request_total{job="kube-apiserver",verb="GET",code="200"}'
+    values: '0 10 20 50 90 99'
+  - series: 'apiserver_request_total{job="kube-apiserver",verb="GET",code="500"}'
+    values: '0x2 1x2'
+
+  promql_expr_test:
+  - eval_time: 5m
+    expr: code_verb:apiserver_request_total:increase1h{verb="GET"}
+    exp_samples:
+    - labels: '{__name__="code_verb:apiserver_request_total:increase1h", code="200", verb="GET"}'
+      value: 99.0
+    - labels: '{__name__="code_verb:apiserver_request_total:increase1h", code="500", verb="GET"}'
+      value: 1.0
+
+- name: calculate apiserver request total increase 30d rate
+  interval: 1m
+  input_series:
+  - series: code_verb:apiserver_request_total:increase1h{verb="GET",code="200"}
+    values: '10+10x9'
+  - series: code_verb:apiserver_request_total:increase1h{verb="GET",code="500"}
+    values: '0+1x9'
+
+  promql_expr_test:
+  - eval_time: 10m
+    expr: code_verb:apiserver_request_total:increase30d{verb="GET"}
+    exp_samples:
+    - labels: '{__name__="code_verb:apiserver_request_total:increase30d", code="200", verb="GET"}'
+      value: 3.96e+4
+    - labels: '{__name__="code_verb:apiserver_request_total:increase30d", code="500", verb="GET"}'
+      value: 3.24e+3
```

Comment:
Maybe add a comment on how these values are arrived at? I can understand the tests above at a glance, but not so much with this 30d version - I'm scratching my head a little! 😅

Reply:
I made the calculations using its formula: `avg_over_time(code_verb:apiserver_request_total:increase1h[30d]) * 24 * 30`. I know it's a bit tautological but at least if it changes in the future and the new value isn't backwards compatible we'll know from the failing test

Reply:
Got it, so the full calc for the code 200 series would be:

```math
((10+20+30+40+50+60+70+80+90+100)/10)*24*30
= 39,600
= 3.96*10^4
```